src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -458,8 +458,17 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
             $container->setDefinition(sprintf('%s.definition', $workflowId), $definitionDefinition);
 
             // Add workflow to Registry
-            foreach ($workflow['supports'] as $supportedClass) {
-                $registryDefinition->addMethodCall('add', array(new Reference($workflowId), $supportedClass));
+            if (isset($workflow['supports'])) {
+                foreach ($workflow['supports'] as $supportedClassName) {
+                    $strategyDefinition = new Definition(
+                        Workflow\SupportStrategy\ClassInstanceSupportStrategy::class, array($supportedClassName)
+                    );
+                    $strategyDefinition->setPublic(false);
+                    $registryDefinition->addMethodCall('add', array(new Reference($workflowId), $supportedClassName));
+                }
+            }
+            if (isset($workflow['support_strategy'])) {
+                $registryDefinition->addMethodCall('add', array(new Reference($workflowId), new Reference($workflow['support_strategy'])));
```

Comment:
As a second option, "support_strategy" expects to be a reference to a registered service implementing the introduced SupportStrategyInterface.